core/appmodule/v2/environment.go

```diff
@@ -5,18 +5,21 @@ import (
 	"cosmossdk.io/core/event"
 	"cosmossdk.io/core/gas"
 	"cosmossdk.io/core/header"
+	"cosmossdk.io/core/router"
 	"cosmossdk.io/core/store"
 	"cosmossdk.io/log"
 )
 
 // Environment is used to get all services to their respective module
 type Environment struct {
-	BranchService   branch.Service
-	EventService    event.Service
-	GasService      gas.Service
-	HeaderService   header.Service
+	Logger log.Logger
+
+	BranchService branch.Service
+	EventService  event.Service
+	GasService    gas.Service
+	HeaderService header.Service
+	RouterService router.Router
+
 	KVStoreService  store.KVStoreService
 	MemStoreService store.MemoryStoreService
-	DataBaseService store.DatabaseService
```

Comment:
remove this one?

Reply:
probably better to embed it in environment in v2 and environment here is from v1?

Reply:
I actually never noticed this service before. Seems like it I removed it when resolving a merge conflict.
What this service is for?

Reply:
nonconsensus writes. Ill add it back in a different pr where its wired up

Reply:
Oh, nice!, but is it something that will be available only from store,server/v2?

Shouldn't we strive to keep environment the same for v1 and v2?

Reply:
i believe only store v2 but ill need to check